src/platform/plugins/shared/controls/public/actions/edit_control_display_settings/control_display_settings_popover.tsx

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+import type { EuiSwitchEvent, EuiPopoverProps } from '@elastic/eui';
+import {
+  EuiToolTip,
+  EuiButtonIcon,
+  EuiPopover,
+  EuiWrappingPopover,
+  EuiFormRow,
+  EuiButtonGroup,
+  EuiSpacer,
+  EuiSwitch,
+} from '@elastic/eui';
+import { DEFAULT_CONTROL_GROW, DEFAULT_CONTROL_WIDTH } from '@kbn/controls-constants';
+import type { ControlWidth } from '@kbn/controls-schemas';
+import { i18n } from '@kbn/i18n';
+import { apiCanLockHoverActions } from '@kbn/presentation-publishing';
+import type { PinnableControlApi } from './types';
+import { ACTION_EDIT_CONTROL_DISPLAY_SETTINGS } from '../constants';
+
+interface Props {
+  api: PinnableControlApi;
+  displayName: string;
+  iconType: string;
+}
+
+export const ControlDisplaySettingsPopover: React.FC<Props> = ({ api, displayName, iconType }) => {
+  // When the user changes a setting and the new layout is applied, the popover gets destroyed and re-rendered
+  // Make sure it reopens if the user hasn't closed it
+  const isPopoverOpenInitialState = useMemo(
+    () => (apiCanLockHoverActions(api) ? api.hasLockedHoverActions$.value : false),
+    [api]
+  );
+
+  const [isPopoverOpen, setIsPopoverOpen] = useState(isPopoverOpenInitialState);
+
+  const layoutState = api.parentApi.layout$.value;
+  const layoutEntry = useMemo(() => layoutState.controls[api.uuid], [layoutState, api.uuid]);
+  const isToRightOfGrowControl = useMemo(
+    () => layoutEntry.order > 0 && Object.values(layoutState.controls)[layoutEntry.order - 1].grow,
+    [layoutEntry.order, layoutState.controls]
+  );
```

Comment:
We use this to determine where to pin the popover. If we are **not** to the right of a `grow` control, we can pin the popover to the control label rather than the action button so that it doesn't jump around as things change; otherwise, we can't avoid the popover jumping around as things change, so we just pin it to the button as normal.

| `isToRightOfGrowControl` | `!isToRightOfGrowControl` |
|--------|--------|
| ![Dec-11-2025 09-33-03](https://github.com/user-attachments/assets/e41462cf-33ea-4163-bfb9-74b8f08f7f44) | ![Dec-11-2025 09-30-24](https://github.com/user-attachments/assets/f704d120-5f4a-4056-a2c2-d41d4a3b2748) |